docs/application/native/guides/multimedia/image-edit.md

```diff
@@ -321,111 +437,67 @@ To decode a JPEG, PNG, GIF, or BMP image:
-
-   ```
-   unsigned long long size = 0; /* Encoded image size */
-   ret = image_util_encode_run(encode_h, &size);
-   ```
+   > Due to the encoder limitations, the color space setting is only supported for encoding the JPEG images. The default color space is `IMAGE_UTIL_COLORSPACE_RGBA8888`. PNG, GIF, and BMP images are encoded with `IMAGE_UTIL_COLORSPACE_RGBA8888`.
 
-7. After the encoding is complete, destroy the encoding handle using the `image_util_encode_destroy()` function:
+4. After the encoding is complete, destroy the encoding handle using `image_util_encode_destroy()`:
 
    ```
    ret = image_util_encode_destroy(encode_h);
    ```
+
 <a name="animation"></a>
 ## Encoding an Animated GIF
 
 To encode an animated GIF image:
 
-1. Create an encoding handle using the `image_util_encode_create()` function:
+1. Create an encoding handle using `image_util_agif_encode_create()`:
 
    ```
-   image_util_type_e encoder_type = IMAGE_UTIL_GIF;
-   image_util_encode_h encode_h = NULL;
-   ret = image_util_encode_create(encoder_type, encode_h);
+   image_util_agif_encode_h agif_encode_h = NULL;
+   ret = image_util_agif_encode_create(&agif_encode_h);
    ```
 
-2. For each frame:
-
-   1. Set the image to the input buffer using the `image_util_encode_set_input_buffer()` function:
-
-      ```
-      ret = image_util_encode_set_input_buffer(encode_h, buffer);
-      ```
-
-   2. Set the width and height of the input buffer using the `image_util_encode_set_resolution()` function:
-
-      ```
-      ret = image_util_encode_set_resolution(encode_h, width, height);
-      ```
-
-   3. Set the delay time between GIF frames using the `image_util_encode_set_gif_frame_delay_time()` function:
-
-      ```
-      ret = image_util_encode_set_gif_frame_delay_time(encode_h, buffer);
-      ```
-
-3. To save the encoded image, set the output path or buffer using the `image_util_encode_set_output_path()` or `image_util_encode_set_output_buffer()` function:
+2. Add the images with the delay time between GIF frames using `image_util_agif_encode_add_frame()`:
 
    ```
-   ret = image_util_encode_set_output_path(encode_h, path);
+   ret = image_util_agif_encode_add_frame(agif_encode_h, src_image, delay_time);
    ```
 
-4. Execute the encoding using the `image_util_encode_run()` function:
+3. Save the encoded image using `image_util_agif_encode_save_to_file()` or `image_util_agif_encode_save_to_buffer()`:
 
    ```
-   unsigned long long size = 0; /* Animated GIF image size */
-   ret = image_util_encode_run(encode_h, &size);
+   ret = image_util_agif_encode_save_to_file(agif_encode_h, path);
    ```
 
-5. After the encoding is complete, destroy the encoding handle using the `image_util_encode_destroy()` function:
+4. After the encoding is complete, destroy the encoding handle using `image_util_encode_destroy()`:
 
    ```
-   ret = image_util_encode_destroy(encode_h);
+   ret = image_util_agif_encode_destroy(agif_encode_h);
```

Comment:
`image_util_encode_destroy()` in description, `image_util_agif_encode_destroy()` in sample code.
Would you please check?

Reply:
okay, `image_util_encode_destroy()` should be changed to `image_util_agif_encode_destroy()` in description. I will replace it.